taurus_pyqtgraph/datainspectormodel.py

```diff
@@ -0,0 +1,205 @@
+
+        self.plot = plot
+
+        self.scatter = {}
+        self.informationClouds = {}
+        # Maybe good idea is give user's access
+        # to they have possibility to change the cloud style?
+        self.cloudStyle = "background-color: #35393C;"
+        self.custom_point_style_init = False
+
+        self.date_format = "%Y-%m-%d %H:%M:%S"
+
+        self.v_line = InfiniteLine(angle=90, movable=False)
+        self.plot.addItem(self.v_line, ignoreBounds=True)
+
+    def __del__(self):
+        """
+        Destructor which removing all data inspector item's
+        """
+        self.detach()
+
+    def mouseMoved(self, evt):
+        """
+        Slot use to handle the mouse move event, and preform the action
+        on the plot.
+
+        :param evt: mouse event
+        """
+        pos = evt[0]
+
+        if self.plot.sceneBoundingRect().contains(pos):
+            mousePoint = self.plot.vb.mapSceneToView(pos)
+
+            # Move InfiniteLine to the new position
+            self.v_line.setPos(mousePoint.x())
+
+            # Iteration over each curves in the plot and checking
+            # that mouse point if is near the x axis date
+            for i, curve in enumerate(filter(lambda o: not isinstance(o, TaurusTrendSet),
+                                             self.plot.curves)):
+
+                curve_data = curve.getData()
+                curve_x_data = curve_data[0]
+                curve_y_data = curve_data[1]
+
+                # Getting the nearest point in the x axis
+                # (if exist - empty array if not)
+                point_index = self.getPointIndex(curve_x_data, mousePoint.x())
+                if len(point_index) > 0:
+
+                    point_x = curve_x_data[point_index][0]
+                    point_y = curve_y_data[point_index][0]
+                    try:
+                        # Painting the cloud with the values of
+                        # the selected data and blink the point base
+                        # on the i values which is the index of current curve
+                        self.paintCloud(point_x, point_y, i)
+                        self.triggerPoint(curve, point_index, i)
+
+                    except KeyError:
+
+                        # Create the TextItem object for current curve and init
+                        # the point style
+                        self.informationClouds[i] = TextItem()
+
+                        parent = self.plot.curves[i]
+                        self.informationClouds[i].setParentItem(parent)
+
+                        self.scatter[i] = curve.scatter
+                        if curve.opts['symbol'] is None:
+                            # create the custom style of the point
+                            # if point style (symbol and size) not exist
+                            curve.setSymbol('o')
+                            curve.setSymbolSize(10)
+                            self.custom_point_style_init = True
```

Comment:
this flag (`self.custom_point_style_init`) should be stored *per curve*. Otherwise, if a plot contains 2 curves, one having symbols and the other not having them, the symbols will be erased for **all** the curves when calling `reset_custom_style`